lib/galaxy/tool_util/parser/xml.py

```diff
@@ -10,6 +10,7 @@
 
 import packaging.version
 
+from galaxy.authnz.util import provider_name_to_backend
```

Comment:
Can you think of a way to avoid this import here and ideally also in evaluation.py ?
We want to be able to reason about tools in these modules without considering the server configuration, and this is breaking the package tests.

I guess one way would be to do the `provider_name_to_backend` check when building the concret tool class in https://github.com/mvdbeek/galaxy/blob/7eb3260e589c28cc046a606375355c76aa0df653/lib/galaxy/tools/__init__.py#L563

Reply:
I see what you mean - tool validation should not depend on server config. I removed that part, only checking now that the syntax is correct:  `oidc_[id|access|refresh]_token_xxx`. If we cannot expose a token, we just set the corresponding environment variable to `token-unavailable`. This would happen now if a) `xxx` is not configured in Galaxy (or misspelled in a tool file) or b) a user has not logged in via this provider. 

For evaluation.py - I'm not sure how can we avoid calling `provider_name_to_backend ` to map the provider name to the backend name (e.g. `azure`->`azuread-v2-tenant-oauth2`). We could request to use the backend name in a tool XML file, but this is probably not a good idea. Or we could change the scheme and store the provider name in 
the database entry for OIDC tokens. Not sure if that is a good idea as well. But without any of these two options we need this mapping.

Reply:
The code in evaluation.py may run on the compute node not having a database available, but we do build the and serialize the environment variables while we still have a database connection, so that's actually fine for now. We may later move this into the JobWrapper class, but it's fine for now.